MonumentAddons.cs

```diff
@@ -5320,6 +5328,18 @@ private void DisableVehicleDecay(BaseEntity vehicle)
                     return;
                 }
             }
+
+            private void SetHackableCrateDecay(HackableLockedCrate hackableCrate)
+            {
+                if (hackableCrate == null)
+                    return;
+
+                if (hackableCrate.shouldDecay || !hackableCrate.HasFlag(HackableLockedCrate.Flag_FullyHacked))
+                    return;
```

Comment:
My tendency is to cancel the timer after the condition is met, rather than simply doing nothing in the method call. That could be achieved by doing the following. Additionally, invokes should generally call `TrackStart()` and `TrackEnd()` on the plugin to expose the CPU cost to users.

```cs
Action setDecay = null;
setDecay = () =>
{
    _pluginInstance?.TrackStart();
    if (hackableCrate.HasFlag(HackableLockedCrate.Flag_FullyHacked))
    {
        hackableCrate.shouldDecay = true;
        hackableCrate.RefreshDecay();
        hackableCrate.CancelInvoke(setDecay);
    }
    _pluginInstance?.TrackEnd();
};
hackableCrate.InvokeRepeating(setDecay, UnityEngine.Random.Range(0f, 5f), 300f);
```

Reply:
I couldn't figure out how to cancel annonymous invoke. Using 'hack complete' hook to enable decay seemed so wrong, so I rather let it run until decayed. Introducing `Action` variable is so obvious I'm ashamed